src/pip/_internal/resolution/resolvelib/resolver.py

```diff
@@ -148,14 +148,13 @@ def resolve(self, root_reqs, check_supported_wheels):
                 # doesn't really allow for the logger.critical calls I'm
                 # using here.
                 for req, parent in e.causes:
+                    if parent is None:
+                        req_disp = str(req)
+                    else:
+                        req_disp = '{} (from {})'.format(req, parent.name)
                     logger.critical(
-                        "Could not find a version that satisfies " +
-                        "the requirement " +
-                        str(req) +
-                        ("" if parent is None else " (from {})".format(
-                            parent.name
-                        ))
-                    )
+                        "Could not find a version that satisfies "
+                        "the requirement %s", req_disp)
```

Comment:
I wonder if this would work:

```python
if parent is None:
    message = (
        "Could not find a version that satisfies "
        "the requirement {}"
    )
else:
    message = (
        "Could not find a version that satisfies "
        "the requirement {} (from {})"
    )
logger.critical(message, req, parent.name)
```

It’d make the log record much cleaner IMO.

Reply:
After applying GH-8435 to enable `str.format` it should work, but the default logger does not support `str.format` just yet.

Reply:
Cool, looking forward to it. m

Reply:
I've just tried this and realized that in this case we still need to check for `parent` to pass `parent.name`.  BTW other than this is there anything else that should be changed?